src/core/blocks/serialization.cpp

```diff
@@ -361,6 +361,8 @@ struct LoadImage : public FileBase {
 };
 
 struct WritePNG : public FileBase {
+  std::vector<uint8_t> _scratch;
```

Comment:
Is it necessary to have a field used as buffer? Can't it be a temporary variable?

Reply:
Yeah caching, if for some reason this block is called many times there will be no allocations at all (if size <=)

Reply:
Btw this is often done all over the codebase exactly for this reason.